common/src/main/java/com/microsoft/identity/common/internal/cache/MicrosoftStsAccountCredentialAdapter.java

```diff
@@ -153,6 +159,74 @@ public RefreshToken createRefreshToken(
         return refreshToken;
     }
 
+    @Override
+    public Account asAccount(final com.microsoft.identity.common.Account accountIn) {
+        final Account accountOut = new Account();
+
+        if (accountIn instanceof AzureActiveDirectoryAccount) {
+            final AzureActiveDirectoryAccount aadAccount = (AzureActiveDirectoryAccount) accountIn;
+            final IDToken idToken = aadAccount.getIDToken();
+
+            if (null != idToken && null != idToken.getTokenClaims()) {
+                final Map<String, String> idTokenClaims = idToken.getTokenClaims();
+                final String avatarUrl = idTokenClaims.get(IDToken.PICTURE);
+                accountOut.setAvatarUrl(avatarUrl);
+                final String issuerUrlStr = idTokenClaims.get(MicrosoftIdToken.ISSUER);
+
+                if (!StringExtensions.isNullOrBlank(issuerUrlStr)) {
+                    try {
+                        final URL issuerUrl = new URL(issuerUrlStr);
+                        accountOut.setEnvironment(issuerUrl.getHost());
+                    } catch (MalformedURLException e) {
+                        // TODO log a warning
+                    }
+                }
+
+                final String altSecId = idTokenClaims.get("altsecid");
+                accountOut.setGuestId(altSecId);
+            }
+
+            accountOut.setUniqueUserId(aadAccount.getUid() + "." + aadAccount.getUtid());
+            accountOut.setRealm(aadAccount.getTenantId());
+            accountOut.setAuthorityAccountId(aadAccount.getUserId());
+            accountOut.setUsername(aadAccount.getDisplayableId());
+            accountOut.setAuthorityType("AAD"); // TODO What about MSA? How does that work?
+            accountOut.setFirstName(aadAccount.getGivenName());
+            accountOut.setLastName(aadAccount.getFamilyName());
+        }
+
+        if (accountIn instanceof MicrosoftStsAccount) {
+            accountOut.setAuthorityType("MSSTS");
+        }
+
+        // TODO is there a special case needed for ADFS? (authority type)
+        return accountOut;
+    }
+
+    @Override
+    public RefreshToken asRefreshToken(final com.microsoft.identity.common.internal.providers.oauth2.RefreshToken refreshTokenIn) {
+        final RefreshToken refreshTokenOut = new RefreshToken();
+
+        // TODO populate this object
+
+        // Required fields
+        refreshTokenOut.setUniqueUserId("");
+        refreshTokenOut.setEnvironment("");
+        refreshTokenOut.setCredentialType(CredentialType.RefreshToken.name());
+        refreshTokenOut.setClientId("");
+        refreshTokenOut.setSecret(refreshTokenIn.getRefreshToken());
+
```

Comment:
I"d all required fields to the fresh token object.  Base class.... with the exception of credential type....